cnf-certification-test/chaostesting/suite.go

```diff
@@ -0,0 +1,304 @@
+package chaostesting
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/test-network-function/cnf-certification-test/pkg/provider"
+	"github.com/test-network-function/cnf-certification-test/pkg/tnf"
+
+	"github.com/sirupsen/logrus"
+	"github.com/test-network-function/cnf-certification-test/internal/clientsholder"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/test-network-function/cnf-certification-test/cnf-certification-test/identifiers"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	yamlutil "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/restmapper"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/test-network-function/cnf-certification-test/cnf-certification-test/common"
+	"github.com/test-network-function/cnf-certification-test/cnf-certification-test/results"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+)
+
+const (
+	// timeout for eventually call
+	TestCaseTimeout    = 180 * time.Second
+	Deployment         = "deployment"
+	ServiceAccountFile = "chaostesting/service-account.yaml"
+	ExperimentFile     = "chaostesting/experiment-delete.yaml"
+	chaosEngineFile    = "chaostesting/chaos-engine.yaml"
+	chaosname          = "pod-delete"
+	completedResult    = "completed"
+	pass               = "Pass"
+)
+
+var _ = ginkgo.Describe(common.ChaosTesting, func() {
+	var env provider.TestEnvironment
+
+	ginkgo.BeforeEach(func() {
+		env = provider.GetTestEnvironment()
+	})
+	ginkgo.ReportAfterEach(results.RecordResult)
+
+	// container security context: privileged escalation
+	testID := identifiers.XformToGinkgoItIdentifier(identifiers.TestPodDeleteIdentifier)
+	ginkgo.It(testID, ginkgo.Label(testID), func() {
+		testPodDelete(&env)
+	})
+
+})
+
+func testPodDelete(env *provider.TestEnvironment) {
+	for _, dep := range env.Deployments {
+		namespace := dep.Namespace
+		label := labelValue(env, dep.Spec.Template.Labels)
+		if err := applyAndCreateFiles(label, Deployment, namespace); err != nil {
+			ginkgo.Fail(fmt.Sprintf("test failed while creating the files %s", err))
+		}
+		completed := waitForTestFinish(TestCaseTimeout)
```

Comment:
You can use the same style as below. Namely, `if completed := waitForTestFinish(TestCaseTimeout); !completed {`